### PR TITLE
Ropes issue6

### PR DESCRIPTION
**Pull request by dweinflash**
Add functionality to delete images from the database, using a dialogue box to confirm deletion. Deleted images are removed from the home page. Delete links added to the Index and Show pages.

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -108,6 +108,7 @@ def test_show_success
     assert_equal Image.last.url, VALID_IMAGE_URL
     assert_equal Image.last.tag_list, [valid_tag]
     assert_select '.js-image-tag', valid_tag
+    assert_select '.js-delete-link', count: 1
```
Comment (wanyue0714) [inline test/controllers/images_controller_test.rb:111]: Could we also test this destroy link and confirm msg here?

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -28,6 +28,16 @@ def create
     end
   end
 
+  def destroy
+    @image = Image.find_by(id: params[:id])
+    if @image
+      @image.destroy
+    else
+      flash[:error] = 'Cannot delete image - it does not exist!'
```
Comment (wanyue0714) [inline app/controllers/images_controller.rb:36]: 👍 for the edge case

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -3,6 +3,33 @@
 VALID_IMAGE_URL = 'https://www.gstatic.com/webp/gallery3/1.png'.freeze
 
 class ImagesControllerTest < ActionDispatch::IntegrationTest
+  def test_delete_from_db
+    image_delete = Image.create!(url: VALID_IMAGE_URL, tag_list: 'delete')
+    assert_difference 'Image.count', -1 do
+      delete image_path(image_delete.id)
+    end
+  end
```
Comment (wanyue0714) [inline test/controllers/images_controller_test.rb:11]: We don't test db in controller_tests, normally controller_tests just cover controller action logic, and view changes

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -3,6 +3,33 @@
...
+
+  def test_delete_fail_image_dne
+    new_id = 1
+    get image_path(new_id)
+    assert_includes response.body, 'No image found!'
+    assert_no_difference 'Image.count' do
+      delete image_path(new_id)
+    end
+    assert_equal 'Cannot delete image - it does not exist!', flash[:error]
```
Comment (wanyue0714) [inline test/controllers/images_controller_test.rb:30]: This test is good! Could we also add one more line here to assert the view redirect to index page?

File: test/controllers/images_controller_test.rb [hunk 5]
```diff
@@ -96,6 +96,7 @@ def test_index
     assert_select 'tr', 2
     assert_select 'td', 'tag2'
     assert_select 'td', 'tag1'
+    assert_select '.js-delete-link', count: 2
```
Comment (wanyue0714) [inline test/controllers/images_controller_test.rb:99]: same here https://github.com/AppFolioOnboarding/image-sharer-dweinflash/pull/14/commits/f523359fca7e4691a859a9fd0dff60bfd7139bb8#r448019832
